components/script/dom/document.rs

```diff
@@ -596,6 +600,45 @@ impl Document {
         }
     }
 
+    /// https://html.spec.whatwg.org/multipage/#scroll-to-the-fragment-identifier
+    pub fn check_and_scroll_fragment(&self, fragment: &str) {
+        let target = self.find_fragment_node(fragment);
+
+        // Step 1
+        self.set_target_element(target.r());
+
+        let point = if fragment.is_empty() || fragment.to_lowercase() == "top" {
```

Comment:
Darnnnnn, suddenly found that I should just use `Cell` for `ServoUrl` rather than `DOMRefCell`. Sadface :-(